### ISSUE TITLE
Mesh Faces are out of order when Transparency is on

### ISSUE THREAD

**Issue by LittleDuck702** [godotengine/godot#50794]
### Godot version

3.3.2

### System information

MacOS 11.3.1, M1, GLES3

### Issue description

When the “transparent” flag is turned on in a material and “Cull Mode” is disabled, the meshes faces are out of order.

<img width="909" alt="Screen Shot 2021-07-23 at 9 14 09 PM" src="https://user-images.githubusercontent.com/87885710/126855287-67006e29-31d7-41da-bb82-2cd144be5410.png">

### Steps to reproduce

Create a mesh and a material, add a texture and set the “Transparent” flag to true, also you will need to set “Cull Mode” to disable.

### Minimal reproduction project

[BugReport.zip](https://github.com/godotengine/godot/files/6871809/BugReport.zip)

**Comment by Calinou** [comment]
This is expected due to the limitationd of transparency sorting. As a workaround, use alpha scissor (alpha testing) instead of alpha blending.

**Comment by LittleDuck702** [comment]
When I turn the “transparent“ flag off and turn on alpha scissors, something really odd happens. Is this normal?

<img width="909" alt="IMG_9735" src="https://user-images.githubusercontent.com/87885710/126856497-82210c34-548a-415d-8ac5-51c9089f8bce.png">

**Comment by Calinou** [comment]
If the depth draw mode is set to Opaque Prepass, this could be due to https://github.com/godotengine/godot/issues/36669.

**Comment by LittleDuck702** [comment]
Yep, I have Opaque Pre-Pass is turned on and #36669 looks exactly like what is happening here.

**Comment by LittleDuck702** [comment]
One of the work around in #36669 did work, but sadly the cube’s faces are still out of order.

<img width="909" alt="Screen Shot 2021-07-23 at 11 18 48 PM" src="https://user-images.githubusercontent.com/87885710/126857101-b9ba3179-7959-46f5-9d62-3844db2a36b0.png">

**Comment by Calinou** [comment]
> One of the work around in #36669 did work, but sadly the cube’s faces are still out of order.

To prevent this starting in 3.4beta, you can render each face with its own MeshInstance using a PlaneMesh or QuadMesh resource.. This relies on https://github.com/godotengine/godot/pull/50721, so it won't work in 3.3.2.

**Comment by LittleDuck702** [comment]
Oh okay, where can I download the beta, Or do I have to compile it from source?

**Comment by Calinou** [comment]
> Oh okay, where can I download the beta, Or do I have to compile it from source?

You have to [compile the `3.x` branch from source](https://docs.godotengine.org/en/stable/development/compiling/), as there are no beta releases yet.

**Comment by LittleDuck702** [comment]
Okay thanks, I’ll go test that.

**Comment by Calinou** [comment]
@WowieMan Did you manage to get it working in [3.4 beta6](https://godotengine.org/article/dev-snapshot-godot-3-4-beta-6)?

**Comment by WowieMan** [comment]
No, not really. I was able to get ￼ around it by creating two meshes, one for the shadow and one for the ￼ visuals, though that's probably not the best ￼solution ￼long-term.￼
